client/src/pages/Signup/components/SignupForm.tsx

```diff
@@ -0,0 +1,279 @@
+
+//=============== interfaces ===============//
+interface SignupFields {
+  username: string;
+  password: string;
+  confirmPassword: string;
+  securityQuestion: string;
+  securityAnswer: string;
+}
+
+interface SignupTextFieldProps {
+  id: string;
+  label: string;
+  value: string;
+  type?: string;
+  onChange: (event: React.ChangeEvent<HTMLInputElement>) => void;
+  onBlur: (event: React.FocusEvent<HTMLInputElement>) => void;
+  error?: boolean;
+  helperText?: string;
+}
+
+//=============== functions ===============//
+function delay(ms: number) {
+  return new Promise((resolve) => setTimeout(resolve, ms));
+}
+
+//=============== consts ===============//
+const validationSchema = yup.object().shape({
+  username: yup.string().required('Username cannot be empty'),
+  password: yup
+    .string()
+    .min(8, 'Password must be at least 8 characters')
+    .required('Password cannot be empty'),
+  confirmPassword: yup
+    .string()
+    .oneOf([yup.ref('password'), ''], 'Passwords must match')
+    .required('Confirm password cannot be empty'),
+  securityQuestion: yup.string().required('Security question cannot be empty'),
+  securityAnswer: yup
+    .string()
+    .required('Security answer answer cannot be empty'),
+});
+
+const SignupTextField: React.FC<SignupTextFieldProps> = ({
+  id,
+  label,
+  value,
+  type = 'text',
+  onChange,
+  onBlur,
+  error,
+  helperText,
+}) => {
+  return (
+    <TextField
+      required
+      id={id}
+      label={label}
+      variant="filled"
+      size="small"
+      type={type}
+      inputProps={{
+        style: { color: theme.palette.primary.contrastText },
+      }}
+      InputLabelProps={{
+        style: { color: theme.palette.primary.contrastText },
+      }}
+      style={{ width: 300 }}
+      value={value}
+      onChange={onChange}
+      onBlur={onBlur}
+      error={error}
+      helperText={helperText}
+    />
+  );
+};
+
+/**
+ * Signup Textfield
+ * @returns {JSX.Element} - Signup page
+ */
+const SignupForm = (): JSX.Element => {
+  const navigate = useNavigate();
+  const [usernameError, setUsernameError] = React.useState(' ');
+  const [generalError, setGeneralError] = React.useState(false);
+  const [openSnackbar, setOpenSnackbar] = React.useState(false);
+
+  const handleSignup = async (values: SignupFields) => {
+    try {
+      const { username, password, securityQuestion, securityAnswer } = values;
+      const response: AxiosResponse = await signup({
+        username,
+        password,
+        securityQuestion,
+        securityAnswer,
+      });
+      if (response.status === 201) {
+        setOpenSnackbar(true);
+        await delay(3000);
+        navigate('/login');
+      } else {
+        setGeneralError(true);
+      }
+    } catch (error) {
+      if (axios.isAxiosError(error)) {
+        const axiosError: AxiosError = error;
+        if (axiosError.response?.status === 400) {
+          setUsernameError('Username already exists');
```

Comment:
This is fine for now as long as we're careful when we change the API in the future

Reply:
I was battling with it some time to match the message of the response but did not have any luck